pkg/arm/v11/resources.go

```diff
@@ -558,11 +606,14 @@ func vmss(cs *api.OpenShiftManagedCluster, app *api.AgentPoolProfile, backupBlob
 				DiskSizeGB:   to.Int32Ptr(256),
 			},
 		}
-		(*(*vmss.VirtualMachineProfile.NetworkProfile.NetworkInterfaceConfigurations)[0].VirtualMachineScaleSetNetworkConfigurationProperties.IPConfigurations)[0].PublicIPAddressConfiguration = &compute.VirtualMachineScaleSetPublicIPAddressConfiguration{
-			Name: to.StringPtr(armconst.VmssNicPublicIPConfigurationName),
-			VirtualMachineScaleSetPublicIPAddressConfigurationProperties: &compute.VirtualMachineScaleSetPublicIPAddressConfigurationProperties{
-				IdleTimeoutInMinutes: to.Int32Ptr(15),
-			},
+
+		if !cs.Properties.PrivateAPIServer {
+			(*(*vmss.VirtualMachineProfile.NetworkProfile.NetworkInterfaceConfigurations)[0].VirtualMachineScaleSetNetworkConfigurationProperties.IPConfigurations)[0].PublicIPAddressConfiguration = &compute.VirtualMachineScaleSetPublicIPAddressConfiguration{
```

Comment:
this, and the removal of the public LB rules, will both affect the monitoring script - it will hang waiting for these IPs to be populated - will need an update.

Reply:
I know, will do. i'll post a todo list in the description